frontend/src/components/filter/Filter.tsx

```diff
@@ -15,173 +14,186 @@ import {API_URL} from '../../constants';
+  //  function for adding categories to filteritem
+  const addCategory = (categoryData: any) => {
+    Object.keys(categoryData).map((categoryName: string, index: number) =>
+      filterItem.push(
+          {id: index.toString(), value: categoryName, isChecked: false},
+      ));
+    setStatus({checklist: filterItem});
+    return categoryData;
   };
   useEffect(() => {
-    // const fetchData = async () => {
-    fetch(`${API_URL}/tags`)
+    fetch(`${API_URL}/categories`)
         .then((res) => res.json())
-        .then((data) => setTags(taglist(data)));
-    setStatus(tagitem);
+        .then((categoryData) => setCategoryData(addCategory(categoryData)));
+    if (categoryData) {
+      (Object.keys(categoryData)).map((category) => {
+        return tempObj.category = false;
+      });
+    }
+    setCheckBoxStatus(tempObj);
+
     // eslint-disable-next-line
   }, []);
-  // / dummy function for refreshing component
-  const ddff = () => {
-    setX((x) => x + 1);
+
+  // fetch api function
+  const fetchApi = (typeurl: string, verifiedurl: string, categoryurl: any) => {
+    fetch(`${API_URL}/resources/${typeurl}/${verifiedurl}?tags=${categoryurl} `)
+        .then((resp) => resp.json())
+        .then((data) => {
+          const taskarr = data.sort((first: any, second: any) => {
+            if (first.name > second.name) {
+              return 1;
+            } else {
+              return -1;
+            }
+          });
+          store.dispatch(
+              {
+                type: FETCH_TASK_SUCCESS,
+                payload: taskarr,
+              },
+          );
+        });
+  };
+  // / function for showing types
+  let typeIcon: any;
+  const addIcon = (it: any, idx: number) => {
+    typeIcon = idx === 0 ? <BuildIcon
+      size="sm" color="black"
+      style={{marginLeft: '-0.5em'}} /> :
+      <DomainIcon size="sm"
+        color="black"
+        style={{marginLeft: '-0.5em'}} />;
+  };
+
+
+  // custom label for type filter
+  const customLabel = (typeName: string) => {
+    return <Flex>
+      <FlexItem breakpointMods={[{modifier: FlexModifiers['spacer-xs']}]}>
+        {typeIcon}
+      </FlexItem>
+      <FlexItem>
+        {typeName}
+      </FlexItem>
+    </Flex>;
   };
-  //  formation of filter url for fetching task and pipelines
-  const filterurl = (e: any) => {
+
+
+  // formation of filter url  for calling filterAPi to
+  //  fetching task and pipelines
+  const filterApi = (event: any) => {
     let typeurl = 'all';
     let verifiedurl = 'all';
-    let tagsurl = '';
-
-    status.forEach((it: any) => {
-      if (it.id === e.target.id) {
-        it.isChecked = e.target.checked;
+    let categoryurl = '';
+    const target = event.target;
+    // for handling isChecked parameter of checkbox
+    setCheckBoxStatus({...checkBoxStatus, [target.value]: target.checked});
+    status.checklist.forEach((it: any) => {
+      if (it.id === event.target.id) {
+        it.isChecked = event.target.checked;
```

Comment:
return soon after this line? do you expect to find more than 1 item matching the `id` ?

Reply:
no,expect 1 item matching the id.